contrib/exporters/secadvisor/mod/acc_rej_action.go

```diff
@@ -0,0 +1,231 @@
+	}
+	if f.Flow1.Metric.BAPackets > f.Flow2.Metric.BAPackets {
+		newMetricReported.BAPackets = f.Flow2.Metric.BAPackets
+	}
+	if f.Flow1.Metric.BABytes > f.Flow2.Metric.BABytes {
+		newMetricReported.BABytes = f.Flow2.Metric.BABytes
+	}
+	// Note: Metric.Start should never change
+	if f.Flow1.Metric.Last > f.Flow2.Metric.Last {
+		newMetricReported.Last = f.Flow2.Metric.Last
+	}
+	// compute LastUpdateMetric based on previous MetricReported and current newMetricReported
+	var newLastUpdateMetric flow.FlowMetric
+	if f.Status == FlowInfo_OLD_FLOW {
+		newLastUpdateMetric.ABPackets = newMetricReported.ABPackets - f.MetricReported.ABPackets
+		newLastUpdateMetric.ABBytes = newMetricReported.ABBytes - f.MetricReported.ABBytes
+		newLastUpdateMetric.BAPackets = newMetricReported.BAPackets - f.MetricReported.BAPackets
+		newLastUpdateMetric.BABytes = newMetricReported.BABytes - f.MetricReported.BABytes
+		newLastUpdateMetric.Start = f.MetricReported.Last
+		newLastUpdateMetric.Last = newMetricReported.Last
+	} else {
+		// this is the first report for the flow
+		newLastUpdateMetric = *f.Flow1.LastUpdateMetric
+		if f.Flow1.LastUpdateMetric.ABPackets > f.Flow2.LastUpdateMetric.ABPackets {
+			newLastUpdateMetric.ABPackets = f.Flow2.LastUpdateMetric.ABPackets
+		}
+		if f.Flow1.LastUpdateMetric.ABBytes > f.Flow2.LastUpdateMetric.ABBytes {
+			newLastUpdateMetric.ABBytes = f.Flow2.LastUpdateMetric.ABBytes
+		}
+		if f.Flow1.LastUpdateMetric.BAPackets > f.Flow2.LastUpdateMetric.BAPackets {
+			newLastUpdateMetric.BAPackets = f.Flow2.LastUpdateMetric.BAPackets
+		}
+		if f.Flow1.LastUpdateMetric.BABytes > f.Flow2.LastUpdateMetric.BABytes {
+			newLastUpdateMetric.BABytes = f.Flow2.LastUpdateMetric.BABytes
+		}
+		newLastUpdateMetric.Last = newMetricReported.Last
+	}
+	newF.Metric = &newMetricReported
+	newF.LastUpdateMetric = &newLastUpdateMetric
+
+	// FIXME - What should we do with UpdateCount?
+	newF.Action = "ACCEPT"
+	f.MetricReported = newMetricReported
+	f.LastUpdateMetricReported = newLastUpdateMetric
+	return &newF
+}
+
+// CleanupOldEntries - delete old entries from table; allocate a new table and preserve only the live connections
+func (t *accRejAction) CleanupOldEntries() {
+	// allocate a new status buffer
+	accRejStateNew := make(map[string]*accRejFlowInfo)
+	for _, f := range t.accRejState {
+		if f.Status == FlowInfo_ONE_ENTRY { // a Rejected flow
+			continue
+		}
+		if f.Flow1.FinishType != flow.FlowFinishType_NOT_FINISHED { // flow was finished
+			continue
+		}
+		L3TrackingID := f.Flow1.GetL3TrackingID()
+		accRejStateNew[L3TrackingID], _ = t.accRejState[L3TrackingID]
+	}
+	t.accRejState = accRejStateNew
+	now := time.Now()
+	t.timeOfLastCleanup = now.Unix()
+}
+
+// AccRejAction - combines flows from 2 measurement points to determine whether a flow was Accepted or Rejected
+func (t *accRejAction) AccRejAction(fl []*flow.Flow) []*flow.Flow {
+	t.InitAccRejAction()
+	t.AccumulateAccRejInfo(fl)
```

Comment:
take the state out of `t` and make is a local variable

Reply:
No. We need the state across calls to this function.

Reply:
ok